elasticdl/python/master/k8s_instance_manager.py

```diff
@@ -225,9 +225,19 @@ def _event_cb(self, event):
             # Workaround for memory leak issues in tf eager mode.
             # A pod may fail due to OOM from tf eager mode memory leak.
             failed_pod = False
-            if evt_type == "MODIFIED" and phase == "Failed":
+            if (
+                evt_type == "MODIFIED"
+                and phase == "Failed"
+                and evt_obj.status.container_statuses
+                and evt_obj.status.container_statuses[0].state.terminated
+                and evt_obj.status.container_statuses[
+                    0
+                ].state.terminated.reason
+                == "OOMKilled"
```

Comment:
Why only this reason?

Reply:
If other reasons, there must be bugs in the code/model, and no need to relaunch.